jwplatform/upload.py

```diff
@@ -0,0 +1,132 @@
+import os
+from enum import Enum
+from hashlib import md5
+import requests
+from requests import HTTPError
+
+from jwplatform import constants
+
+
+class UploadType(Enum):
+    direct = "direct"
+    multipart = "multipart"
+
+
+def determine_upload_method(file, target_part_size) -> str:
+    filename = file.name
+    file_size = os.stat(filename).st_size
+    if file_size <= target_part_size:
+        return UploadType.direct.value
+    return UploadType.multipart.value
+
+
+class MultipartUpload:
+
+    def __init__(self, client, upload_id: str, file, target_part_size, retry_count):
+        self.upload_id = upload_id
+        self.target_part_size = target_part_size
+        self.upload_retry_count = retry_count
+        self.file = file
+        self.client = client
+
+    def upload(self):
+
+        if self.target_part_size < constants.MIN_PART_SIZE:
+            raise ValueError(f"The part size has to be at least greater than {constants.MIN_PART_SIZE} bytes.")
+
+        filename = self.file.name
+        file_size = os.stat(filename).st_size
+        part_count = file_size // self.target_part_size + 1
```

Comment:
We should probably handle the case where the part_count is over 10,000, since that will likely happen for files >50GB (our goals is to eventually handle files up to 250GB)